### PR TITLE
Model builder tests

### PR DESCRIPTION
**Pull request by michaelraczycki**
Extending the model_builder test suite

### CODE REVIEW

**Comment by michaelraczycki** [comment]
@twiecki , let me know if you'd do something differently in the unit tests, I've decided to go first with full tests before trying to integrate with baseMMM class

File: pymc_experimental/tests/test_model_builder.py [hunk 1]
```diff
@@ -129,6 +149,12 @@ def test_predict():
     assert isinstance(pred["y_model"][0], float)
 
 
+def test_predict_posterior():
+    data, model_config, sampler_config = test_ModelBuilder.create_sample_input()
+    model = test_ModelBuilder(model_config, sampler_config, data)
+    model.fit()
+
```
Comment (twiecki) [inline pymc_experimental/tests/test_model_builder.py:156]: should have an assert here, even if it's just a simple one.

Comment (michaelraczycki) [reply pymc_experimental/tests/test_model_builder.py:162]: I've added them, and while adding I've noticed that I'd be introducing 4 or 5 repetitions of the same few lines so threw in a helper function, take a look at it in free moment and let me know if you like the end result.

File: pymc_experimental/tests/test_model_builder.py [hunk 2]
```diff
@@ -99,9 +105,7 @@ def test_fit():
     sys.platform == "win32", reason="Permissions for temp files not granted on windows CI."
 )
 def test_save_load():
-    data, model_config, sampler_config = test_ModelBuilder.create_sample_input()
-    model = test_ModelBuilder(model_config, sampler_config, data)
-    model.fit()
+    model = test_ModelBuilder.initial_build_and_fit(False)
```
Comment (twiecki) [inline pymc_experimental/tests/test_model_builder.py:108]: Nice refactor!

**Review by twiecki — APPROVED** [review]

**Comment by twiecki** [comment]
Thanks 👍

**Comment by michaelraczycki** [comment]
Glad to help

**Pull request merged**